backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/parser/NonContainmentReferenceHandler.java

```diff
@@ -55,32 +60,48 @@ public NonContainmentReferenceHandler(MessageReporter messageReporter) {
     public boolean isNonContainmentReference(final Element owner, String referenceName, final JsonNode astValue) {
         if (astValue.isContainerNode()) {
             JsonNode referenceTypeNode = astValue.get("$type");
-            JsonNode textNode = astValue.get("text");
-            return referenceTypeNode != null && astValue.has("reference") && textNode != null && textNode.isTextual();
+            return referenceTypeNode != null && this.referenceNodeTester.test(astValue);
         }
         return false;
     }
 
     public void createProxy(final Element owner, String referenceName, final JsonNode astValue) {
-        JsonNode referenceTypeNode = astValue.get("$type");
-        JsonNode textNode = astValue.get("text");
-        String referenceType = referenceTypeNode.asText();
-        final EObject refrenceTypeInstance = this.typeBuilder.createObject(astValue);
-
-        if (refrenceTypeInstance instanceof InternalEObject internalTarget) {
-            String qualifiedNameTarget = textNode.asText();
-            internalTarget.eSetProxyURI(URI.createGenericURI("syson-import", QUALIFIED_CONST, qualifiedNameTarget));
-            // It should be a reference
-            Optional<EReference> optEReference = this.referenceComputer.getNonContainmentReference(owner.eClass(), refrenceTypeInstance.eClass(), referenceName);
-            if (optEReference.isEmpty()) {
-                this.messageReporter.error(MessageFormat.format("Unable to find a reference from {0} to {1} with reference name {2} on object {3}", owner.eClass().getName(),
-                        refrenceTypeInstance.eClass().getName(), referenceType, this.logNameProvider.getName(owner)));
+        if (this.isMetadataAccessExpressionCase(owner, referenceName, astValue)) {
+            /**
+             * Handle special case of MetadataAccessExpression.ReferencedElement that have changed between 2024-12 and
+             * 2025-04.
+             * <p>
+             * In 2024-12, this is a true EReference from MetadataAccessExpression to an element. In 2025-04, this a
+             * derived EReference computed from a ownedMembership. As a consequence the current version of SysIDE import
+             * that information as a reference and not has SysML structure with membership. This special case, aims to
+             * provide a workaround by creating the required Membership inside the MetadataAccessExpression and setting
+             * the proxy from MetadataAccessExpression.ReferencedElement to Membership.memberElement.
+             * </p>
```

Comment:
Please add a comment stipulating that this should be removed/reworked when the current version of SysIDE will be at least 2025-04.

Reply:
Done.